src/providers/WorkflowCore.Persistence.EntityFramework/Services/EventsPurger.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using WorkflowCore.Interface;
+using WorkflowCore.Persistence.EntityFramework.Interfaces;
+using WorkflowCore.Persistence.EntityFramework.Models;
+
+namespace WorkflowCore.Persistence.EntityFramework.Services
+{
+    public class EventsPurger : IEventsPurger
+    {
+        private readonly IWorkflowDbContextFactory _contextFactory;
+
+        public EventsPurger(IWorkflowDbContextFactory contextFactory)
+        {
+            _contextFactory = contextFactory;
+        }
+
+        /// <summary>
+        /// Event table purger
+        /// </summary>
+        /// <param name="olderThan"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task PurgeEvents(DateTime olderThan, CancellationToken cancellationToken = default)
+        {
+            var olderThanUtc = olderThan.ToUniversalTime();
+            using (var db = ConstructDbContext())
+            {
+                var events = db.Set<PersistedEvent>()
+                    .Where(x => x.EventTime < olderThanUtc &&
+                                x.IsProcessed == true);
+
+                db.RemoveRange(events);
```

Comment:
@viktorshevchenko210 Since the WorkflowCore uses EF Core 7 now, can we use EF Core 7 bulk delete feature here?

Reply:
This will most likely time out with the default timeout if 30 seconds.  Needs options for timeout in seconds and batch size to fit each specific usage.

Reply:
Mb it makes sense to delete Subscriptions related to this event?

Reply:
@ErikEJ Added options for timeout and batch size.

Reply:
Thanks, Viktor!